### DISCUSSION TITLE
General Help for Updating an old Arena build

### DISCUSSION THREAD

**Discussion by AlexTZHo** [colyseus/colyseus#699] [Q&A]
I am trying to update an old game I made back in 2021 when arena was still around. I would like to self host it in a docker container but all of the old forum posts are gone. Been trying to getting a self-hosted version of the server up and running but there are large holes in my understanding of how Colyseus works.  Should I update it to the current code or make it work as a self-hosted server with the old 0.14.17 Colyseus? 

It currently still runs fine on Localhost. I think I'm narrowing it down to trying to figure out self-hosted route. I just wanted see what others thought and if anyone had any advice with something like this.

**Comment by endel** [comment]
Hi @gurulantern 👋  you can check the migration guide here: https://docs.colyseus.io/migrating/0.15/

Version 0.14 is much harder to scale due to the reliance on the custom proxy. So I'd recommend upgrading if you can. If you don't need scaling self-hosting 0.14 would be fine too
